packages/perspective/index.d.ts

```diff
@@ -0,0 +1,149 @@
+declare module '@jpmorganchase/perspective' {
+    /**** object types ****/
+    export enum TypeNames {
+        STRING = 'string',
+        FLOAT = 'float',
+        INTEGER = 'integer',
+        BOOLEAN = 'boolean',
+        DATE = 'date'
+    }
+
+    export type ValuesByType = {
+        [ key in TypeNames ]: Array<string>
+    }
+
+    export type ValueByType = {
+        TypeNames: Array<string>
+    }
+
+    export enum SortOrders {
+        ASC = 'asc',
+        ASC_ABS = 'asc abs',
+        DESC = 'desc',
+        DESC_ABS = 'desc abs',
+        NONE = 'none',
+    }
+
+    enum NUMBER_AGGREGATES {
```

Comment:
These might potentially be better as Enums rather than strings?

Reply:
they are enums, or you mean to create a global enum with all of them and NUMBER, STRING, BOOLEAN to be just a subset of them?

Reply:
I was thinking that maybe we can take the aggregate keywords and Enum those. But I couldn't find a nice way to make them all align up. Since these are private I'll take another look later and we can hoist them into private enums later.

Reply:
thanks